### PR TITLE
add metrics for IPFS timeouts and other things - PLAT-849

### PR DESCRIPTION
**Pull request by gvelez17**
## Description

We want to see charts of IPFS timeouts, other errors, and also pin success/failure

## How Has This Been Tested?

Tests pass

## PR checklist

Before submitting this PR, please make sure:

- [ x] I have tagged the relevant reviewers and interested parties
- [ ] I have updated the READMEs of affected packages
- [ ] I have made corresponding changes to the documentation

## References:

Please list relevant documentation (e.g. tech specs, articles, related work etc.) relevant to this change, and note if the documentation has been updated.

### CODE REVIEW

File: packages/core/src/dispatcher.ts [hunk 1]
```diff
@@ -238,11 +241,12 @@ export class Dispatcher {
           console.warn(
             `Timeout error while loading CID ${asCid.toString()} from IPFS. ${retries} retries remain`
           )
+          Metrics.count(METRIC_NAMES.IPFS_TIMEOUT, 1)
           if (retries > 0) {
             continue
           }
         }
-
+
```
Comment (smrz2001) [inline packages/core/src/dispatcher.ts:249]: Can revert this.

File: packages/cli/src/ceramic-daemon.ts [hunk 2]
```diff
@@ -816,6 +816,7 @@ export class CeramicDaemon {
     const streamId = StreamID.fromString(req.params.streamid || req.params.docid)
     const { force } = req.body
     await this.ceramic.pin.add(streamId, force)
+    Metrics.count(METRIC_NAMES.STREAM_PINNED, 1)
```
Comment (smrz2001) [inline packages/cli/src/ceramic-daemon.ts:819]: Some of these flows are synchronous API calls, will the metrics code add any latency?

**Review by smrz2001 — APPROVED** [review]
LGTM apart from a couple of minor comments.

File: packages/cli/src/ceramic-daemon.ts (hunk 2 above)
Comment (gvelez17) [reply packages/cli/src/ceramic-daemon.ts:819]: it should not - it should be counting internally and the exporter able to be scraped.  it does not have to do a network call to count.

**Pull request merged**